src/app/clusters/soil-measurement-server/soil-measurement-cluster.cpp

```diff
@@ -0,0 +1,120 @@
+/*
+ *    Copyright (c) 2025 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "soil-measurement-cluster.h"
+#include <app/reporting/reporting.h>
+#include <clusters/SoilMeasurement/Metadata.h>
+#include <lib/core/CHIPError.h>
+#include <tracing/macros.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+SoilMeasurementCluster::SoilMeasurementCluster(EndpointId endpointId) : DefaultServerCluster({ endpointId, SoilMeasurement::Id })
+{
+    mSoilMoistureMeasuredValue.SetNull();
+}
+
+CHIP_ERROR SoilMeasurementCluster::Startup(ServerClusterContext & context)
+{
+    ReturnErrorOnFailure(DefaultServerCluster::Startup(context));
+
+    return CHIP_NO_ERROR;
+}
+
+DataModel::ActionReturnStatus SoilMeasurementCluster::ReadAttribute(const DataModel::ReadAttributeRequest & request,
+                                                                    AttributeValueEncoder & encoder)
+{
+    switch (request.path.mAttributeId)
+    {
+    case SoilMeasurement::Attributes::SoilMoistureMeasurementLimits::Id: {
```

Comment:
consider adding a `using namespace chip::app::Clusters::SoilMeasurement::Attributes` and simplify several of these longer constants.

Remove extra `{}` throughout